inspirehep/ext/cache_manager/manager.py

```diff
@@ -0,0 +1,147 @@
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import logging
+
+from werkzeug.utils import import_string
+
+from .utils import CachedInvalidator, JinjaCacheInvalidator, MemoizedInvalidator
+
+logger = logging.getLogger(__name__)
+
+
+class InspireCacheManager(object):
+    """Class designed to invalidate different types of cache (all types provided by Flask Cache)
+     with Redis as the cache backend"""
+
+    def __init__(self, app=None, cache=None, views={}):
+        """
+            Parameters
+        ----------
+        app : Flask application object
+        cache : Flask Cache object
+        views : dict
+            Views being cached by cache (Flask Cache object). See the 'views' dict sample below:
+            {
+                'main_page': {
+                    'type': 'cached',
+                    'function_reference': 'app.views.main_view'
+                },
+
+                'view_with_have_calculations': {
+                    'type': 'memoize',
+                    'function_reference': 'app.views.extremely_heavy_view'
+                },
+
+                'view_done_with_jinja': {
+                    'type': 'jinja_cache',
+                    'template_name': 'jinja_view',
+                    'kwargs': ['view_ID']
+                }
+            }
+
+            Each view should have 'type' of either 'cached', 'memoize' or 'jinja_cache' value
+            (three types of cache provided by Flask Cache).
+
+            For 'cached' and 'memoized' the field 'function_reference' has to be also defined.
+
+            For 'jinja_cache' two extra fields need to be defined:
+                - 'template_name' which is the name associated with the cached template
+                - 'kwargs' is the list of key_arguments names used to parametrize cached template
+                    e.g. view_id, index etc.
+
+        """
+        self.jinja_invalidator = JinjaCacheInvalidator(cache)
+        self.memoized_invalidator = MemoizedInvalidator(cache)
+        self.cached_invalidator = CachedInvalidator(cache)
+        self.cache_client = cache
+        self.app = app
+        self.views = None
+        if app is not None:
+            self.init_app(app)
+
+    @staticmethod
+    def import_function_by_string(func_string):
```

Comment:
This is already supported by Werkzeug's `import_string`: just use `:` as the last delimiter.